x-pack/plugins/fleet/server/routes/agent/upgrade_handler.ts

```diff
@@ -158,22 +160,40 @@ export const getCurrentUpgradesHandler: RequestHandler = async (context, request
   }
 };
 
-export const checkKibanaVersion = (version: string, kibanaVersion: string) => {
+export const checkKibanaVersion = (version: string, kibanaVersion: string, force = false) => {
   // get version number only in case "-SNAPSHOT" is in it
   const kibanaVersionNumber = semverCoerce(kibanaVersion)?.version;
   if (!kibanaVersionNumber) throw new Error(`kibanaVersion ${kibanaVersionNumber} is not valid`);
   const versionToUpgradeNumber = semverCoerce(version)?.version;
   if (!versionToUpgradeNumber)
     throw new Error(`version to upgrade ${versionToUpgradeNumber} is not valid`);
 
-  if (semverGt(versionToUpgradeNumber, kibanaVersionNumber))
+  if (!force && semverGt(versionToUpgradeNumber, kibanaVersionNumber)) {
     throw new Error(
       `cannot upgrade agent to ${versionToUpgradeNumber} because it is higher than the installed kibana version ${kibanaVersionNumber}`
     );
+  }
+
+  if (
+    force &&
+    !(
+      semverMajor(kibanaVersionNumber) > semverMajor(versionToUpgradeNumber) ||
```

Comment:
I think that this condition would be more readable if extracted in a function. This file is becoming kinda difficult to read with all the different cases, simplifying ot would help maintaining it :)

Reply:
Agreed it's confusing :) I extracted this into named variables and added comments. I didn't think a function would make a lot of sense because it's not reused elsewhere. Let me know what you think.